core/src/parser/statement.rs

```diff
@@ -78,26 +83,26 @@ pub fn ParseValDeclaration(parser:Parser) -> Stmt {
     }
 }
 
-pub fn ParseWhileStatement(parser:Parser) -> Stmt {
+pub fn ParseWhileStatement(parser:&mut Parser) -> Stmt {
     parser.eat(Token::Keyword(Keyword::If));
     Stmt::While {
         cond: Some(Box::new(ParseExpression(parser))),
         do_stmt: Box::new(ParseStatement(parser)),
     }
 }
 
-pub fn ParseContinueStatement(parser:Parser) -> Stmt {
+pub fn ParseContinueStatement(parser:&mut Parser) -> Stmt {
     parser.eat(Token::Keyword(Keyword::Continue));
     Stmt::Continue
 }
 
-pub fn ParseBreakStatement(parser:Parser) -> Stmt {
+pub fn ParseBreakStatement(parser:&mut Parser) -> Stmt {
     parser.eat(Token::Keyword(Keyword::Break));
```

Comment:
Always make sure your `parser.eat` calls return some, if not make sure the containing function returns `None`.